spec/classes/all_roles_spec.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+# Copyright (c) 2018 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+require 'spec_helper'
+require_relative '../support/contexts/with_mocked_nodes'
+
+def puppet_role_name_from(path)
+  path.strip.gsub('/', '::').gsub(%r{^manifests}, 'nebula').gsub(%r{\.pp}, '')
+end
+
+`find manifests/role -name '*.pp'`.each_line do |file_path|
+  role_name = puppet_role_name_from(file_path)
+  describe role_name do
+    on_supported_os.each do |os, os_facts|
+      context "on #{os}" do
+        let(:facts) { os_facts }
+
+        let(:hiera_config) do
+          if role_name.start_with? 'nebula::role::webhost::htvm'
+            'spec/fixtures/hiera/hathitrust_config.yaml'
+          elsif role_name.start_with? 'nebula::role::hathitrust'
+            'spec/fixtures/hiera/hathitrust_config.yaml'
+          elsif role_name.start_with? 'nebula::role::tools_lib'
+            'spec/fixtures/hiera/tools_lib_config.yaml'
+          elsif role_name.start_with? 'nebula::role::chipmunk'
+            'spec/fixtures/hiera/chipmunk_config.yaml'
+          elsif role_name.start_with? 'nebula::role::deb_server'
+            'spec/fixtures/hiera/deb_server_config.yaml'
+          elsif role_name.start_with? 'nebula::role::deploy_host'
+            'spec/fixtures/hiera/named_instances_config.yaml'
+          else
+            'spec/fixtures/hiera/default_config.yaml'
+          end
+        end
+
+        let!(:puppetdb_query) do
+          MockFunction.new('puppetdb_query') do |f|
+            f.stubbed.returns([])
+          end
+        end
+
+        it do
+          begin
+            is_expected.to compile
+          rescue RSpec::Expectations::ExpectationNotMetError => e
+            # Append a message with a suggestion of how to fix the role.
+            x = RSpec::Expectations::ExpectationNotMetError.new(
+              "You probably need to add a hiera_config to spec/classes/all_roles_spec.rb:\n #{e.message}",
+            )
```

Comment:
This is a bit ugly, but at least it'll point to where the error is happening, hopefully with some clarity.

Reply:
We could do this in perhaps a little bit cleaner way by subclassing the `compile` matcher (https://github.com/rodjek/rspec-puppet/blob/master/lib/rspec-puppet/matchers/compile.rb) and overriding `failure_message` and `failure_message_when_negated`

Reply:
I think this is way better.